intro/solutions/pi_wallis.py

```diff
@@ -2,6 +2,8 @@
 The correction for the calculation of pi using the Wallis formula.
 """
 from __future__ import division
+from __future__ import print_function
+from functools import reduce
```

Comment:
Maybe you can move this import in the "cell" where it is needed.

Note how this code uses "notebook-style"! Once we get sphinx-gallery in, we'll have the benefice immediately.